secondarylib-feedback/src/main/scala/com/socrata/datacoordinator/secondary/feedback/FeedbackContext.scala

```diff
@@ -116,18 +116,22 @@ class FeedbackContext[CT,CV](user: String,
                              rows: IndexedSeq[Row[CV]],
                              targetColumns: Set[UserColumnId]): Either[ComputationFailure, Map[Int, Map[UserColumnId, CV]]] = {
     val perDatasetData = computationHandler.setupDataset(cookie)
-    val perColumnData = cookie.strategyMap.toSeq.collect {
+    val perColumnDataTargetColMap = cookie.strategyMap.toSeq.collect {
       case (targetCol, strat) if targetColumns.contains(targetCol) & computationHandler.matchesStrategyType(strat.strategyType) =>
-        computationHandler.setupColumn(perDatasetData, strat, targetCol)
-    }
+        (computationHandler.setupColumn(perDatasetData, strat, targetCol) -> targetCol)
+    }.toMap
+    val perColumnData = perColumnDataTargetColMap.keys.toSeq
     val toCompute = rows.iterator.zipWithIndex.flatMap { case (row, index) =>
       val rcis =
         perColumnData.flatMap { columnData =>
           // don't compute if there has been to change to the source columns
-          if (noChange(row, columnData.strategy.sourceColumnIds))
+          val targetColId = perColumnDataTargetColMap(columnData)
+          // New value of either target or source will have the same existing if update does not contain the field.
+          // There should be no change either they contain the same value or are not passed in
+          if (noChange(row, columnData.strategy.sourceColumnIds :+ targetColId))
```

Comment:
refactor

Reply:
That doesn't change the thing I was talking about, which was the `columnData.strategy.sourceColumnIds :+ targetColId` part.  This will always produce the same result for a given column, but constructs a new version of it for each row in the column.